vue/components/ui/molecules/scrollable-items/scrollable-items.vue

```diff
@@ -0,0 +1,246 @@
+            default: null
+        },
+        /**
+         * Enables/Disables the previous/next arrow buttons that are shown in
+         * the left and right side of component.
+         */
+        arrows: {
+            type: Boolean,
+            default: true
+        },
+        /**
+         * If set to true, the arrows will disappear when there is enough space
+         * for all the items to be visible.
+         */
+        autoHideArrows: {
+            type: Boolean,
+            default: true
+        },
+        /**
+         * Dictates how the items are horizontally aligned. Example: "center".
+         */
+        alignment: {
+            type: String,
+            default: "center"
+        },
+        /**
+         * Dictates the horizontal scroll speed of the items when using the mouse
+         * scroll wheel. Negative values inverts the default direction of the scroll.
+         */
+        scrollSpeed: {
+            type: Number,
+            default: 10
+        },
+        /**
+         * Props for the arrow button icon components.
+         */
+        buttonIconProps: {
+            type: Object,
+            default: () => ({})
+        }
+    },
+    data: function() {
+        return {
+            selectedData: this.selected,
+            scrollableWidth: 0
+        };
+    },
+    computed: {
+        itemsContainerStyle() {
+            return {
+                "text-align": this.alignment
+            };
+        },
+        _buttonIconProps() {
+            return {
+                borderRadius: 0,
+                iconStrokeWidth: 3,
+                ...this.buttonIconProps
+            };
+        },
+        showArrows() {
+            const hideArrows = this.autoHideArrows && this.scrollableWidth <= 0;
+            return this.arrows && !hideArrows;
+        },
+        selectedIndex() {
+            return this.items.findIndex(item => item.value === this.selectedData);
+        },
+        isPreviousDisabled() {
+            return this.selectedIndex <= 0;
+        },
+        isNextDisabled() {
+            return this.selectedIndex >= this.items.length - 1;
+        }
+    },
+    watch: {
+        items() {
+            this.$nextTick(() => this.calculateScrollableWidth());
+        },
+        selected(value) {
+            this.selectedData = value;
+        },
+        selectedData(value) {
+            this.$nextTick(() => this.snapSelectedToCenter());
+            this.$emit("update:selected", value);
+        }
+    },
+    created: function() {
+        window.addEventListener("resize", this.onResize);
+    },
+    destroyed: function() {
+        window.removeEventListener("resize", this.onResize);
+    },
+    methods: {
+        itemClasses(item, index) {
+            const base = {};
+            base[`item-${item.value}`] = true;
+            if (item.value === this.selectedData) base.selected = true;
+            return base;
+        },
+        calculateScrollableWidth() {

```

Comment:
Make it clear in the method name that this mutates internal state. Suggestions: updateScrollableWidth, refreshScrollableWidth, ...